api-controllers/ServiceFabrikApiController.js

```diff
@@ -47,6 +48,39 @@ class ServiceFabrikApiController extends FabrikBaseController {
     super();
   }
 
+  validateUuid(uuid, description) {
+    const uuidPattern = /^[0-9a-f]{8}(?:-[0-9a-f]{4}){3}-[0-9a-f]{12}$/i;
+    if (!uuidPattern.test(uuid)) {
+      throw new errors.BadRequest(`Invalid ${description || 'uuid'} '${uuid}'`);
+    }
+  }
+
+  setPlan(req) {
+    if (req.plan === undefined) {
+      return Promise
+        .try(() => {
+          const plan_id = req.body.plan_id || req.query.plan_id;
+          if (plan_id) {
+            this.validateUuid(plan_id, 'Plan ID');
+            return plan_id;
+          }
+          const instance_id = req.params.instance_id;
+          assert.ok(instance_id, 'Middleware assignManager requires a plan_id or instance_id');
+          return cf.cloudController
```

Comment:
I would say let us create an issue to check if we can make plan id mandatory so that we do not need to go to CF. 

CC @amitmalav

Reply:
Agree. Let's revisit all broker apis and see if we can mandate `plan_id`

Reply:
https://github.com/cloudfoundry-incubator/service-fabrik-broker/issues/448